### ISSUE TITLE
Introduce Emails Activities

### ISSUE THREAD

**Issue by Bonapara** [twentyhq/twenty#3133]
# Email is received

### New email received

![CleanShot 2023-12-22 at 15 22 15](https://github.com/twentyhq/twenty/assets/19412894/76db862f-61ab-482a-8feb-41139a3438c4)

### New reply received

![CleanShot 2023-12-22 at 15 23 08](https://github.com/twentyhq/twenty/assets/19412894/ca2bca68-3da1-4469-8463-fda8a54072e3)

# Email is sent

### Reply sent

![CleanShot 2023-12-22 at 15 23 25](https://github.com/twentyhq/twenty/assets/19412894/1fc706c5-ddb0-47d9-b367-f04770775e62)

### New email sent

![CleanShot 2023-12-22 at 15 23 42](https://github.com/twentyhq/twenty/assets/19412894/00f0aa4c-f081-4776-9c19-94a6968e80c4)

# Email preview

Clicking the `Chevron down` icon button expands the email content

## Full sharing

![CleanShot 2023-12-22 at 15 24 42](https://github.com/twentyhq/twenty/assets/19412894/de22feb4-8e15-4918-9105-14494222b8b8)

## Subject and metadata are shared

![CleanShot 2023-12-22 at 15 26 07](https://github.com/twentyhq/twenty/assets/19412894/01993ecf-6201-4b16-b026-1d8456d769ce)

## Metadata only are shared

![CleanShot 2023-12-22 at 15 26 32](https://github.com/twentyhq/twenty/assets/19412894/81558ed8-e56c-4886-be2b-d4aa59cf69b3)

**Comment by FelixMalfait** [comment]
@Eyesonly88 Merry Christmas to you too! 
That's 100% on the roadmap but we're not there yet, we're going there step-by-step:
First step is to import emails from gmail automatically, that should be live in January.
Second step is the ability to reply to emails through the gmail API, that should be live in February.
And third step is to move from the "1-to-1 sales email" use-case to a more "marketing / mass email sender" use-case. That's going to take a few more months to do it well probably.

**Comment by cocobeach** [comment]
IMO I think we could do 2 birds with 1 stone, if we implement the call straight into an Openai custom Assistant's actions API scheme, so everyone can set it up and customise how it wants.`
Including adding  some knowledge based gathered from sent emails (but I would leave that feature manual and individual as it needs to be done only once to "teach" the AI to answer the right things on the basis of similar past answers.
Otherwise we only follow what other CRM do and don't differentiate ourselves, like that we solve the problem of sending and receiving emails, and we get it done automatically or semi automatically?

**Comment by FelixMalfait** [comment]
@cocobeach 100% agree AI assistants will be super cool to implement, along with embeddings of emails, in the context of "Automations" which is a workstream we will tackle later this year. But mail is really foundational for CRM so it makes sense to have proper gmail/office/smtp integrations first. AI is only as powerful as the integration you give it - we need to get that structure right first!

**Comment by cocobeach** [comment]
Agreed, in fact I can tell you from experience , having worked with and for Salesforce for the last 18 years, the best implementations are the ones you never need to open.

For a CRM data entry is the pain, sales people are lazy - takes one to know one - and the more you data-entry automatically the more the CRM will be up to date and useful.
Automation needs to accomplish jobs not creating alerts that stagnate in the system. Now with the AI you can do high quality data entry, oops scoring, forecasting, reporting etc...

Having smtp is great but having it intercepted and entered in the CRM by the AI is the biggest differentiator you can think of.

It's because for some reason my API is not recognised - prolly because I am on local host.

But you can extract the schema of the CRM and use it inside the actions of the assistant in json or yaml, then in system prompt you put the various actions for the AI to recognise and use.
(Next step is di tu e the AI on the actions themselves).

My 2 cents...

Ps if you use OvidijusParsiunas/deep-chat you can probably include it in the interface very easily and offer an array of AI solutions from local to OpenAI Assistants out of the box, with everything else in the middle. 

;)
